src/haystack/analyzer.clj

```diff
@@ -338,14 +338,25 @@
           (flag-duplicates)
           (flag-tooling)))))
 
+(defn- compile-like-exception?
```

Comment:
I think you should add some of your explanations from the PR here, so it's clear why we need this.